layer/logged_data/queuing_network_log_destination.py

```diff
@@ -0,0 +1,78 @@
+import queue
+import threading
+from queue import Queue
+from typing import Any, Callable, List, Optional
+
+from layer.clients.logged_data_service import LoggedDataClient
+from layer.logged_data.data_logging_request import DataLoggingRequest
+from layer.logged_data.log_data_destination import LogDataDestination
+from layer.logged_data.retrying_files_sender import RetryingFilesSender
+
+
+WAIT_INTERVAL_SECONDS = 1
+
+
+class QueueingNetworkLogDestination(LogDataDestination):
+    def __init__(self, client: LoggedDataClient) -> None:
+        self._log_data_client = client
+        self._files_storage = RetryingFilesSender()
+
+        self._sending_errors: List[Any] = []
+        self._stop_reading = False
+        self._local_queue: Queue[DataLoggingRequest] = Queue()
+        self._reading_thread = threading.Thread(target=self._send_elem_from_queue)
+        self._reading_thread.start()
+
+    def enqueue(
+        self,
+        func: Callable[[LoggedDataClient], Optional[Any]],
+        data: Optional[Any] = None,
+    ) -> None:
+        self._local_queue.put_nowait(
+            DataLoggingRequest(
+                self._files_storage, lambda: func(self._log_data_client), data
+            )
+        )
+
+    def close(self) -> None:
+        if not self._stop_reading:
+            self._stop_reading = True
+            self._reading_thread.join(timeout=3 * WAIT_INTERVAL_SECONDS)
```

Comment:
Why the `3 *`?

Reply:
you're right, that was a bad idea, changed to explicit timeout as described in description